stream_alert/rule_processor/threat_intel.py

```diff
@@ -13,90 +13,389 @@
+            mapping_str (str): A qualified string has pattern 'normalized_type:ioc_type'
+
+        Returns:
+            (bool): First return indicate if the string a qualifited string contains
+                both normalized CEF type and IOC type.
+            (str): Second return is normalized type.
+            (str): Last return is IOC type.
         """
-        if cls.__intelligence:
-            return
-        if (config.get('threat_intel')
-                and config['threat_intel'].get('enabled')
-                and config['threat_intel'].get('mapping')):
-            cls.__intelligence = cls.read_compressed_files(intel_dir)
-            cls.__config = config['threat_intel'].get('mapping')
+        normalized_type = None
+        ioc_type = None
+
+        splitted_str = mapping_str.split(':')
+        if len(splitted_str) == 1:
+            normalized_type = splitted_str[0]
+        elif len(splitted_str) == 2:
+            normalized_type = splitted_str[0]
+            ioc_type = splitted_str[1].split('_')[-1]
+        else:
+            LOGGER.info('Key %s in conf/types.json is incorrect', mapping_str)
+            return False, None, None
+
+        if normalized_type and ioc_type:
+            return True, normalized_type, ioc_type
+
+        return False, normalized_type, None
+
+    def _process_ioc(self, ioc_collections):
+        """Check if any info is malicious by querying DynamoDB IOC table
+
+        Args:
+            ioc_collections (list): A list of StreamIoc instances.
+        """
+        # Segment data before calling DynamoDB table with batch_get_item.
+        for subset in StreamThreatIntel._segment(ioc_collections):
+            query_values = []
+            for ioc in subset:
+                if ioc.value not in query_values:
+                    query_values.append(ioc.value)
+            query_result = self._query(query_values)
+            for value in ioc_collections:
+                for ioc in query_result:
+                    if value.value.lower() == ioc[PRIMARY_KEY]:
+                        value.itype = ioc[SUB_TYPE_KEY]
+                        value.is_ioc = True
+                        continue
+
+    @staticmethod
+    def _segment(ioc_collections):
+        """Static method to segment ioc_collections in to smaller set.
+        Batch query to dynamodb supports up to 100 items.
+
+        Args:
+            ioc_collections (list): A list of StreamIoc instances
+
+        Returns:
+            (list): List of subset of StreamIoc instances
+        """
+        result = []
+        end = len(ioc_collections)
+        for index in range(0, end, MAX_QUERY_CNT):
+            result.append(ioc_collections[index:min(index+MAX_QUERY_CNT, end)])
+        return result
+
+    exceptions_to_backoff = (ClientError)
+    @backoff.on_exception(backoff.expo,
+                          exceptions_to_backoff,
+                          max_tries=3,
+                          giveup=exceptions_to_giveup,
```

Comment:
you could use a `lambda` here like `lambda err: err.response['Error']['Code'] == 'AccessDeniedException'`